developing-with-phonon/secure-backend-specification/README.md

```diff
@@ -0,0 +1,279 @@
+
+### Outputs
+```json
+	[implementation dependent]
+```
+## GENERATE_PHONON
+A phonon keypair will be generated on the phonon sce, and it's public portion will be returned
+### Inputs
+```json
+	{"type":[curve type])
+```
+note: curve type is a string and currently the only option is "secp256k1"
+### Outputs
+```json
+	{"pubkey":"binary data"}
+```
+## SET_DESCRIPTOR
+the descriptor of a phonon contains arbitrary key value pairs and is to be attached to phonons when this command is run. the identifier of the phonon is implementation dependent and may be a representation of the pubkey or an index or other identifier stored on the sce.
+### Inputs
+```json
+	{
+		{
+			phonon:4
+		}
+		"data":{
+		"arbitrary keys":"arbitrary values"
+		}
+	}
+```
+
+### Outputs
+	```json
+	```
+## LIST_PHONONS
+### Inputs
+	```json
+	[optional]
+	```json
+### Outputs
+```json
+	[
+	{
+		"header": {
+			"curve": "secp256k1",
+			"pub": "public key bytes"
+			},
+		"metadata":{
+			"xyz": "abc",
+			"something else":"another value",
+		}
+	]
+```
+note: some implemntations will allow for filter queries to be input to cut down on the amount of phonons returned in a single request
+## SEND_PHONONS
+The sce is given the receiver identity certificate, validates it with the certificate authority, and generates a phonon transfer packet. It is imperative that the phonon is deleted from the sce when this packet is returned to the client. If the receiver certificate cannot be validated, the packet cannot be created. 
+### Encryption
+the output of this command includes encrypted private keys. 
+$ENCRYPTION_SCHEME_DEFINITION (todo)
+### Inputs
+```json
+	{
+		"receiver":"[bytes of full identity certificate of receiver]",
+		"receiver_nonce":56383927,
+		"phonons":[1,3,26]
+		
+	}
+```
+### Outputs
+```json
+	{
+		{
+			"signature":"[signature of the remaining packet signed with identity certificate]",
+			"certificate":"[bytes of full identity certificate of sender]",
+			"counter":56383927,
+			"phonons":[
+				{
+					"header":{
+						"privKey":"[private key bytes encrypted]"
+						"curve": "secp256k1",
+					},
+					"metadata":{
+						"abc":"xyz",
+						"def":"uvw"
+					}
+				}
+			]
+		}
+	}
+```
+## RECEIVE_PHONONS
+receive phonons is run with input of a transfer packet that was output from a sending card's SEND_PHONONS command, When this command is run, the receiving card must verify the legitimacy of the sender card's certificate prior to storage of the phonon. If the verification fails, the packet must not be ingested. The phonon receive counter is then increased to the value specified in the packet.
+### Counter
+the counter is a security measure to prevent a single sce from receiving the same phonon more than once. Upon initialization of the program, the counter must be set to 0, and increases to the value of the most recently received phonon upon ingestion. If a transfer packet comes in with the same or lower counter, the packet must be ignored by the sce. 
```

Comment:
```suggestion
the counter is a security measure to prevent a sce from receiving the same phonon transfer packet more than once. Upon initialization of the program, the counter must be set to 0, and increases to the value of the most recently received phonon upon ingestion. If a transfer packet comes in with the same or lower counter, the packet must be ignored by the sce. 
```